### PR TITLE
Show comment replies

### LINKED ISSUE #2277: Display comment replies
It's great that NewPipe has the feature to view comments since the version 0.16.0! But I miss a function to view the answers of the comments.

### PR DESCRIPTION
**Pull request by xz-dev**
#### What is it?
- [ ] Bugfix (user facing)
- [x] Feature (user facing)
- [ ] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR

Adds a "Show Replies" for the comments with replies for Youtube, just look like youtube.

#### Before/After Screenshots/Screen Record

<table>
  <tr>
    <th>Before</th>
    <th>After1</th>
    <th>After2</th>
    <th>After3</th>
  </tr>
  <tr>
    <td>Before</td>
    <td>After1</td>
    <td>After2</td>
    <td>After3</td>
  </tr>
</table>
Screenshot_20221029-135951_NewPipe Debug

#### Fixes the following issue(s)

- Fixes #2277

#### APK testing

The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by krlvm** [comment]
Can you also set `android:navigationBarColor` to match bottom sheet background color?

**Comment by xz-dev** [comment]
> 

I find something, but I don't know how to fix it.
https://github.com/material-components/material-components-android/issues/267
Maybe we need another professional guy fix it

**Comment by xz-dev** [comment]
> Can you also set `android:navigationBarColor` to match bottom sheet background color?

I find something, but I don't know how to fix it.
https://github.com/material-components/material-components-android/issues/267
Maybe we need another professional guy fix it

**Comment by xz-dev** [comment]
I find there are some code need clean, I'll do latter.

**Review by krlvm — CHANGES_REQUESTED** [review]

File: app/src/main/res/layout/dialog_comment_reply.xml
Comment (xz-dev) [reply app/src/main/res/layout/dialog_comment_reply.xml:11]: You're right, I find it long time.

**Comment by krlvm** [comment]
I don't know if it is possible for me to suggest changes there, I've submitted bottom sheet improvements to @xz-dev fork: https://github.com/xz-dev/NewPipe/pull/1

**Comment by xz-dev** [comment]
> I don't know if it is possible for me to suggest changes there, I've submitted bottom sheet improvements to @xz-dev fork: [xz-dev#1](https://github.com/xz-dev/NewPipe/pull/1)

Of course, thank you

**Comment by krlvm** [comment]
You should not do [it](https://github.com/TeamNewPipe/NewPipe/pull/9020/commits/34233af1bf2576f2a36fc00485e2366e85e922ad).
It is automatically ensured by inheritance.
That's how the current theming strategy works.

**Comment by xz-dev** [comment]
> You should not do [it](https://github.com/TeamNewPipe/NewPipe/pull/9020/commits/34233af1bf2576f2a36fc00485e2366e85e922ad). It is automatically ensured by inheritance. That's how the current theming strategy works.

But it cann't been build, should I use other way to fix it?

**Comment by krlvm** [comment]
It's strange, I'm not getting build errors :/

**Comment by krlvm** [comment]
Sorry, it's been my fault.
Please, revert the last commit and edit following lines

Line 32:
```xml
<item name="bottomSheetDialogTheme">@style/BottomSheetDialogTheme.Light.Base</item>
```

Line 57:
```xml
<item name="bottomSheetDialogTheme">@style/BottomSheetDialogTheme.Dark.Base</item>
```

**Comment by xz-dev** [comment]
> Sorry, it's been my fault. Please, revert the lastest commit and edit following lines
> 
> Line 32:
> 
> ```
> <item name="bottomSheetDialogTheme">@style/BottomSheetDialogTheme.Light.Base</item>
> ```
> 
> Line 57:
> 
> ```
> <item name="bottomSheetDialogTheme">@style/BottomSheetDialogTheme.Dark.Base</item>
> ```

Pushed

**Comment by xz-dev** [comment]
I should go to sleep, and [there](https://github.com/TeamNewPipe/NewPipeExtractor/pull/936) is count support of reply, all code is ready(include NewPipe), hope we can finish it early. :)

**Comment by xz-dev** [comment]
I find a bug that it don't load all reply, I need finger out how reply page work.

**Comment by nicoursi** [comment]
I installed this artifact to test it.  It is nice,  well done. The only thing is that one has to tap on the "show replies" string and is is a bit unconfortable. Could the clickable area be extended at least orizontally? So that,  to get more replies  I could tap on the middle or on the right side and not only on the "show reply" text.

**Comment by opusforlife2** [comment]
> So that, to get more replies I could tap on the middle or on the right side and not only on the "show reply" text.

Rather than that, the Show Replies text could be inlaid in a button with a different colour from the background so that it is clear where to tap.

**Comment by nicoursi** [comment]
It is already clear as it is blue,  but currently the only clickable area in on the left and sometimes is kore comfortable on the right or middle. Same principle of expanding the comment text where you can tap anywhere

**Comment by opusforlife2** [comment]
When you have a scrollable view, it needs some dead zones where no action gets triggered accidentally. Here, for example, you can safely scroll up and down without opening replies when you might not want to.

**Comment by krlvm** [comment]
I would also suggest preventing the bottom sheet from expanding up so that it doesn't overlap the video.

**Comment by xz-dev** [comment]
> 

As [what I say](https://github.com/TeamNewPipe/NewPipe/issues/2277#issuecomment-1256831605), rolling will also overlap the video, video need keep in top.
But, I think it need anther PR to fix it.

**Comment by xz-dev** [comment]
> 

I just imitate Youtube

**Comment by xz-dev** [comment]
> I installed this artifact to test it. It is nice, well done. The only thing is that one has to tap on the "show replies" string and is is a bit unconfortable. Could the clickable area be extended at least orizontally? So that, to get more replies I could tap on the middle or on the right side and not only on the "show reply" text.

I just imitate Youtube

**Comment by xz-dev** [comment]
> I find a bug that it don't load all reply, I need finger out how reply page work.

Fixed https://github.com/TeamNewPipe/NewPipe/pull/9020/commits/7e254c277e7714c5d7c62bcd979c50b1a1b01fc7

**Comment by fGolke** [comment]
Sorry, found another bug: entering and leaving fullscreen mode, after viewing replies on a comment hides all but this comment. Even after reopening the video the other comments keep hidden.
Happens too, when you reopen a video after viewing replies to a comment.
(Build downloaded moments ago, android 9)

**Comment by xz-dev** [comment]
> Sorry, found another bug: entering and leaving fullscreen mode, after viewing replies on a comment hides all but this comment. Even after reopening the video the other comments keep hidden. Happens too, when you reopen a video after viewing replies to a comment. (Build downloaded moments ago, android 9)

Fixed https://github.com/TeamNewPipe/NewPipe/pull/9020/commits/5603e85a605ec0e7b8babe77165b11ccd41310aa

**Comment by tsiflimagas** [comment]
Hey, thank you for working on this! Here's a couple of issues I noticed: after https://github.com/TeamNewPipe/NewPipe/commit/a4cd42bb843283b91ff65f8e2e5c52261e36131a a lot more replies are loaded, but for comments with a lot of replies (like over 200) it reaches a point where no more are loading and you get a buffering animation. Also, when you have replies open and rotate the screen, the app crashes 
## Exception
* __User Action:__ ui error
* __Request:__ ACRA report
* __Content Country:__ US
* __Content Language:__ en-US
* __App Language:__ en_US
* __Service:__ none
* __Version:__ 0.24.0
* __OS:__ Linux Android 13 - 33
<details><summary><b>Crash log </b></summary><p>

```
java.lang.IllegalArgumentException: serviceId is NO_SERVICE_ID
	at org.schabi.newpipe.util.ExtractorHelper.checkServiceId(ExtractorHelper.java:73)
	at org.schabi.newpipe.util.ExtractorHelper.getCommentsInfo(ExtractorHelper.java:177)
	at org.schabi.newpipe.fragments.list.comments.CommentsFragment.loadResult(CommentsFragment.java:107)
	at org.schabi.newpipe.fragments.list.BaseListInfoFragment.startLoading(BaseListInfoFragment.java:141)
	at org.schabi.newpipe.fragments.list.BaseListInfoFragment.doInitialLoadLogic(BaseListInfoFragment.java:115)
	at org.schabi.newpipe.fragments.BaseStateFragment.onViewCreated(BaseStateFragment.java:43)
	at androidx.fragment.app.Fragment.performViewCreated(Fragment.java:3019)
	at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:551)
	at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:261)
	at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:1840)
	at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:1758)
	at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1701)
	at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:2849)
	at androidx.fragment.app.FragmentManager.dispatchViewCreated(FragmentManager.java:2777)
	at androidx.fragment.app.Fragment.performViewCreated(Fragment.java:3020)
	at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:551)
	at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:261)
	at androidx.fragment.app.FragmentStore.moveToExpectedState(FragmentStore.java:113)
	at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1374)
	at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:2841)
	at androidx.fragment.app.FragmentManager.dispatchActivityCreated(FragmentManager.java:2784)
	at androidx.fragment.app.FragmentController.dispatchActivityCreated(FragmentController.java:262)
	at androidx.fragment.app.FragmentActivity.onStart(FragmentActivity.java:478)
	at androidx.appcompat.app.AppCompatActivity.onStart(AppCompatActivity.java:248)
	at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1516)
	at android.app.Activity.performStart(Activity.java:8315)
	at android.app.ActivityThread.handleStartActivity(ActivityThread.java:3701)
	at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:221)
	at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:201)
	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:173)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2308)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7894)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)

```
</details>
<hr>

Furthermore, on the design side, if i may, I'd suggest adding a material divider or a bit of elevation to the header of replies, to separate it a bit from the scrolling view. And for the "Show replies" button, since you've added quite a bit of spacing between it and the next comment, you could make that button bigger and move it closer or totally to the center, horizontally, plus a rectangular ripple when pressing it.

**Comment by xz-dev** [comment]
> but for comments with a lot of replies (like over 200) it reaches a point where no more are loading and you get a buffering animation
Maybe it's a bug of Extractor, another thing is I think we need add [retly count](https://github.com/TeamNewPipe/NewPipeExtractor/issues/935)  first for test it.
> Also, when you have replies open and rotate the screen, the app crashes
Ok, soon
> Furthermore, on the design side, if i may, I'd suggest adding a material divider or a bit of elevation to the header of replies, to separate it a bit from the scrolling view. And for the "Show replies" button, since you've added quite a bit of spacing between it and the next comment, you could make that button bigger and move it closer or totally to the center, horizontally, plus a rectangular ripple when pressing it.
I'm not a professional android APP dev, but I'll try. I welcome you to make it happen yourself, if you want.

**Comment by xz-dev** [comment]
> Could the maintainer devs please approve the workflows so that we can test the upcoming versions of this PR?

Actually, you can try it.
> The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.

**Comment by nicoursi** [comment]
No,  I can't. I get this: " This workflow is awaiting approval from a maintainer in #9020". Only your first commit got approved,  then the next ones need approval,  probably because there ate commits from another 1st committer (krlvm)

**Comment by xz-dev** [comment]
> > Also, when you have replies open and rotate the screen, the app crashes

>Ok, fix soon

I find it's so hard to fix if I use BottomSheetDialogFragment, so I need recode by navigation

**Comment by xz-dev** [comment]
> you could make that button bigger and move it closer or totally to the center, horizontally, plus a rectangular ripple when pressing it.

I agree with https://github.com/TeamNewPipe/NewPipe/pull/9020#issuecomment-1257266844
And also I test it, it's hard to know where is clickable on the right.

**Comment by xz-dev** [comment]
> Furthermore, on the design side, if i may, I'd suggest adding a material divider or a bit of elevation to the header of replies, to separate it a bit from the scrolling view.

> ...plus a rectangular ripple when pressing it

Added

**Comment by tsiflimagas** [comment]
> Added. https://github.com/TeamNewPipe/NewPipe/commit/b91df5e3e81b8a9e89e09d1334648f6dcf6dd1c2 https://github.com/TeamNewPipe/NewPipe/commit/8dda380a5c8d772d88742954020e831cc83cc09d

I was just gonna open a pull request to your fork for these changes. I added the divider just by adding a 1dp thin view under the back button, but toolbar looks even better. Just a little thing, toolbar has some trouble following light theme.

**Comment by xz-dev** [comment]
> > Added. [b91df5e](https://github.com/TeamNewPipe/NewPipe/commit/b91df5e3e81b8a9e89e09d1334648f6dcf6dd1c2) [8dda380](https://github.com/TeamNewPipe/NewPipe/commit/8dda380a5c8d772d88742954020e831cc83cc09d)
> 
> I was just gonna open a pull request to your fork for these changes. I added the divider just by adding a 1dp thin view under the back button, but toolbar looks even better. Just a little thing, toolbar has some trouble following light theme.

I find toolbar theme set text color, so, I remove it to fix that. https://github.com/TeamNewPipe/NewPipe/pull/9020/commits/6eef7150db019bdf4d8ba58e723b5a50544cecf6

**Comment by krlvm** [comment]
"Show replies" button proposition:
- Reduce "Show replies" button padding
- Extend button's width to match parent
- Pick a consistent material color for its text

image

Pull Request: https://github.com/xz-dev/NewPipe/pull/2

**Comment by xz-dev** [comment]
I think the pull request is ready been merge, is there anyone have other questions?

**Review by krlvm — APPROVED** [review]

**Comment by nicoursi** [comment]
Can somebody please approve the workflow so that we can test it?

**Comment by CloverBadland** [comment]
I'm experimenting a great decrease in performance at the moment of loading comment replies, so far just happens when there are many comment replies, I think is better to just load 10 replies and leave to the user the decision to load more replies.

In addition to that, when I am in landscape mode (horizontal) and then I push the button, the window almost don't show:

sc

**Comment by xz-dev** [comment]
> I'm experimenting a great decrease in performance at the moment of loading comment replies, so far just happens when there are many comment replies, I think is better to just load 10 replies and leave to the user the decision to load more replies.

It use pagination loading

**Comment by krlvm** [comment]
I think adding elevation to the titlebar was a questionable choice, it does not really suits the bottom sheet concept

**Comment by xz-dev** [comment]
> I think adding elevation to the titlebar was a questionable choice, it does not really suits the bottom sheet concept

From the beginning, we should not use bottom sheet, a better solution is to use fragment navigation in the comment area, but that requires a lot of changes.

**Comment by gardockt** [comment]
This PR works great - thank you for your effort! Some notes:
- If a video has low comment count, so that they all can fit on a screen with a blank space under them, you cannot scroll comments while holding your finger at the blank space - doing so is possible in current release.
- When looking at replies, I think that tapping someone's avatar to open their channel should close the replies window, since you cannot do anything before closing it. Moreover, if I recall correctly, this is the behavior of the official YouTube app.

**Comment by kyoyacchi** [comment]
Where to download APK?

**Comment by opusforlife2** [comment]
@kyoyacchi Look at the "APK testing" section in the description.

File: app/src/main/java/org/schabi/newpipe/fragments/list/comments/CommentReplyDialog.java
Comment (TobiGr) [inline app/src/main/java/org/schabi/newpipe/fragments/list/comments/CommentReplyDialog.java:83]: cloning the object looks wrong to me. However, I am unable to find a better solution at the moment. @Stypox @AudricV Any opinions or ideas?

**Review by TobiGr — COMMENTED** [review]
Thank you for implementing this long awaited feature!

I am not sure about one thing noted below.

Viewing YouTube and PeerTube (using https://github.com/TeamNewPipe/NewPipeExtractor/pull/941) works well. As you already opened a PR to extract the reply count, you might want to use that info (e.g. instead of "Show replies" use "Show 5 replies"). Introducing and using that plurals string now seems better to avoid creating duplicate effort for the translators.

**Comment by xz-dev** [comment]
Sorry for the relapse of my Mood disorder, I will continue when I have time. I think it should be 1 to 2 weeks later.

**Comment by opusforlife2** [comment]
@xz-dev Take how much ever time you need, mate. No hurries. This is purely voluntary work, after all. Mental health comes first, and programming _much_ later.

**Comment by xz-dev** [comment]
I'm coming back and I make more big change at here.
I remove the dialog, because it looks out of place here.

**Comment by nicoursi** [comment]
Wow,  now it is even faster! 

The only issue i can find so dar is that if i rotate the screen while the replies are open the app crashes.

**Comment by xz-dev** [comment]
> Wow, now it is even faster!
> 
> The only issue i can find so far is that if i rotate the screen while the replies are open the app crashes.

Fixed https://github.com/TeamNewPipe/NewPipe/pull/9020/commits/be3a2420b5312d46899581675da5738a09e21873

**Comment by nicoursi** [comment]
It does not crash with screen rotation but it closes the replies. I think the replies should be kept open with rotation. 

With the new commit,  it says "5 new streams".  I think it should say "5 replies". 

Thanks a lot for your efforts,  you rock!

**Comment by xz-dev** [comment]
> It does not crash with screen rotation but it closes the replies. I think the replies should be kept open with rotation.
> 
> With the new commit, it says "5 new streams". I think it should say "5 replies".
> 
> Thanks a lot for your efforts, you rock!

 LOL, Fixed. https://github.com/TeamNewPipe/NewPipe/commit/f29c941241945aab4a876b51bac7e6347fd4fe5c

File: app/src/main/res/values/strings.xml
Comment (opusforlife2) [reply app/src/main/res/values/strings.xml:135]: > and thus does not need to encourage participation

I don't see how capitalisation or the lack of it could encourage or discourage participation.

But yes, the words should be in lowercase for both scenarios. "1 reply" and "10 replies".

Comment (xz-dev) [reply app/src/main/res/values/strings.xml:135]: I think fully capitalized is because it's a button

Comment (opusforlife2) [reply app/src/main/res/values/strings.xml:135]: Oh, you're saying Android does this automatically?

Comment (xz-dev) [reply app/src/main/res/values/strings.xml:135]: > Oh, you're saying Android does this automatically?

No, it's me manually code, I just think maybe a button text should be fully capitalized? https://github.com/TeamNewPipe/NewPipe/blob/f29c941241945aab4a876b51bac7e6347fd4fe5c/app/src/main/res/layout/list_comments_item.xml#L125

Comment (opusforlife2) [reply app/src/main/res/values/strings.xml:135]: That might be the case normally, but it's not looking good hear. It's somewhat distracting from the actual comments, especially the blue colour.

Try using the regular text colour, and maybe use border/fill colour to highlight that it is a button?

Comment (TobiGr) [reply app/src/main/res/values/strings.xml:135]: I think using blue is ok.

Comment (nicoursi) [reply app/src/main/res/values/strings.xml:135]: I already love it as it is, blue and not capitalized. Blue is good as you can see very well that there are replies for those comments. I agree that capitalized does not look good.

Comment (opusforlife2) [reply app/src/main/res/values/strings.xml:135]: So... let's see a screenshot of blue but not capitalised?

Comment (xz-dev) [reply app/src/main/res/values/strings.xml:135]: Screenshot_20221101-132627_NewPipe Debug

Comment (opusforlife2) [reply app/src/main/res/values/strings.xml:135]: Ah! Perfect! It looks great now.

Comment (xz-dev) [reply app/src/main/res/values/strings.xml:135]: I'll change it like [material 3 say](https://m3.material.io/components/buttons/guidelines#28a1ae9b-9576-44a8-8d2e-8231a0218d20)

**Comment by nicoursi** [comment]
Could you rebase so that you can include the latest exractor fix and we can continue testing your PR? 😀

**Comment by xz-dev** [comment]
> Could you rebase so that you can include the latest exractor fix and we can continue testing your PR? grinning

The exractor is the latest, you can test it. https://github.com/TeamNewPipe/NewPipe/blob/89a1ab52ad1e8f02a0f723943a583d58e0ab855a/app/build.gradle#L190

**Comment by xz-dev** [comment]
> Could you rebase so that you can include the latest exractor fix and we can continue testing your PR? grinning

Updated.

**Comment by nicoursi** [comment]
For some reason this bug happens only on this PR,  it does not happen on the nightlies version. 

To reproduce:
1. Open this channel  https://www.youtube.com/channel/UCZQIPu6LfVfzEnJqDHpWamg
2. Long press and then play in background the video "ATTACCHI CONTRO MEDICI REINTEGRATI"


` ## Exception
* __User Action:__ ui error
* __Request:__ ACRA report
* __Content Country:__ IT
* __Content Language:__ it-IT
* __App Language:__ it_IT
* __Service:__ none
* __Version:__ 0.24.0
* __OS:__ Linux Android 11 - 30
<details><summary><b>Crash log </b></summary><p>

```
java.lang.IllegalArgumentException: No view found for id 0x7f0a0199 (org.schabi.newpipe.debug:id/fragment_container_view) for fragment CommentsFragment{7c055c9} (48d1ff96-556e-4b2a-8b57-c0c6f2a9d30e id=0x7f0a0199)
	at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:513)
	at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:261)
	at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:1840)
	at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:1758)
	at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1701)
	at androidx.fragment.app.FragmentManager$4.run(FragmentManager.java:488)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:223)
	at android.app.ActivityThread.main(ActivityThread.java:7669)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:594)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)

```
</details>
<hr>
`

**Comment by xz-dev** [comment]
> For some reason this bug happens only on this PR, it does not happen on the nightlies version.
> 
> To reproduce:
> 
>     1. Open this channel  https://www.youtube.com/channel/UCZQIPu6LfVfzEnJqDHpWamg
> 
>     2. Long press and then play in background the video "ATTACCHI CONTRO MEDICI REINTEGRATI"
> 
> 
> ` ## Exception
> 
>     * **User Action:** ui error
> 
>     * **Request:** ACRA report
> 
>     * **Content Country:** IT
> 
>     * **Content Language:** it-IT
> 
>     * **App Language:** it_IT
> 
>     * **Service:** none
> 
>     * **Version:** 0.24.0
> 
>     * **OS:** Linux Android 11 - 30
> 
> 
> Crash log
> `

I had download the CI debug APK, I can't reproduce the error.

**Comment by xz-dev** [comment]
> I think the replies should be kept open with rotation.

It's a very basic problem that currently happens to all listing pages, and I don't think I should not fix it at here.

**Comment by AudricV** [comment]
Could you squash the commits with the previous model used into the new one, in order to have a proper branch?

Also, please rebase your PR instead of adding a separate extractor change and use in the future a separate Git branch for your pull requests, like stated in our contribution guidelines.

Thank you in advance.

**Comment by tvnmguy** [comment]
1: 

•Highlight mother comment/main comment's background.

If we select dark theme from settings, then highlight with black tint/gradient.
If we select black theme, then highlight with dark tint/gradient.

•Add total number of comment replies in mother comment/main comment.

•Make mother comment/main comment's icon/avatar little bit larger from comment replies account icon/avatar (Reduce size of comment replies account icon/avatar)

For example👇






For reference;






2:

Adjustment of Replies texts, Mother comment/Main comment, Comment replies, Account icon/avatar for better visual experience and readability 👇








3:

I don't know is this possible, if yes;

Highlight content creator/channel owner comment replies via icon/avatar or like '3 Replies from the hated one and others'

**Review by TobiGr — COMMENTED** [review]
Code looks good. I still don't like the cloning of the comments instance. However, I cannot think of an alternative.
I did not have time to test the changes thoroughly.

**Comment by xz-dev** [comment]
> Could you squash the commits with the previous model used into the new one, in order to have a proper branch?
> 
> Also, please rebase your PR instead of adding a separate extractor change and use in the future a separate Git branch for your pull requests, like stated in our contribution guidelines.
> 
> Thank you in advance.

Ok, move to https://github.com/TeamNewPipe/NewPipe/pull/9410

**Comment by xz-dev** [comment]
> I still don't like the cloning of the comments instance.

I also wan't, but I have no better idea without change Extractor

**Comment by xz-dev** [comment]
> 1:
> 
> •Highlight mother comment/main comment's background.
> 
> If we select dark theme from settings, then highlight with black tint/gradient. If we select black theme, then highlight with dark tint/gradient.
> 
> •Add total number of comment replies in mother comment/main comment.
> 
> •Make mother comment/main comment's icon/avatar little bit larger from comment replies account icon/avatar (Reduce size of comment replies account icon/avatar).
> 
> For examplepoint_down
> 
> 
> 
> 
> 
> For reference;
> 
> 
> 
> 
> 
> 2:
> 
> Adjustment of Replies texts, Mother comment/Main comment, Comment replies, Account icon/avatar for better visual experience and readability point_down
> 
> 
> 
> 
> 
> 
> 
> Advantage: We can eliminate scattered feeling. All (Replies, Mother/Main comment, Comment replies) are in single line.
> 
> 
> 
> 3:
> 
> I don't know is this possible, if yes;
> 
> Highlight content creator/channel owner comment replies via icon/avatar or like '3 Replies from the hated one and others' text.
> 
> 
> 
> 

I think the more important is let it working, if you have more idea about design, you can make a PR to my fork.(it will been included)

**Pull request closed without merging**
